dockerTools: Pin go to 1.9

srhb opened the pull request:
###### Motivation for this change
This is the Go version that Docker currently uses, so I think this is an acceptable way to fix #35505 until the upstream tarsum takes Go 1.10 into account.

###### Things done
Go pinned at 1.9 for dockerTools

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

srhb committed:
dockerTools: Pin go to 1.9

This is the Go version that Docker currently uses:
https://github.com/moby/moby/pull/35739

Fixes #35505

Mic92 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -121,7 +121,7 @@ with pkgs;
   dieHook = makeSetupHook {} ../build-support/setup-hooks/die.sh;
 
-  dockerTools = callPackage ../build-support/docker { };
+  dockerTools = callPackage ../build-support/docker { go = go_1_9; };
```
Can you also downgrade dockers go version and add this link as reference to both? https://github.com/moby/moby/pull/35739

srhb committed:
docker: pin go to 1.9

This is the version used upstream, see:
https://github.com/moby/moby/pull/35739

srhb commented:
@Mic92 Good point. How does that look? :)

srhb commented:
@GrahamcOfBorg test docker

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.35539&attempt_id=b9df7b36-f9e6-4540-bbd8-f607d672e555)

<details><summary>Partial log (click to expand)</summary><p>

```
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.35539&attempt_id=e001cf2c-85bd-4050-a732-4de32d0a4440)

<details><summary>Partial log (click to expand)</summary><p>

```
2018/02/25 17:30:34 WARN: man/src/trust/view.md does not exist, skipping
2018/02/25 17:30:34 WARN: man/src/trust.md does not exist, skipping
2018/02/25 17:30:34 WARN: man/src/volume/prune.md does not exist, skipping
2018/02/25 17:30:34 WARN: man/src/volume/rm.md does not exist, skipping
Generate legacy manpages
post-installation fixup
patching script interpreter paths in /nix/store/s0l0r9300w8fvnjpn7dp7mcrsq3r8kjf-docker-17.12.0-ce
gzipping man pages under /nix/store/by47r4wdrlgilsq61j8dhz6qlz4apllp-docker-17.12.0-ce-man/share/man/
patching script interpreter paths in /nix/store/by47r4wdrlgilsq61j8dhz6qlz4apllp-docker-17.12.0-ce-man
/nix/store/s0l0r9300w8fvnjpn7dp7mcrsq3r8kjf-docker-17.12.0-ce
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.35539&attempt_id=9dd25573-a8a9-46c8-8acf-d5128c8583b5)

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking RPATHs of ELF executables and libraries in /nix/store/j5nj3rs610f8r8cbpa1jv5zls945m8rm-docker-17.12.0-ce
shrinking /nix/store/j5nj3rs610f8r8cbpa1jv5zls945m8rm-docker-17.12.0-ce/libexec/docker/docker
shrinking /nix/store/j5nj3rs610f8r8cbpa1jv5zls945m8rm-docker-17.12.0-ce/libexec/docker/dockerd
patching script interpreter paths in /nix/store/j5nj3rs610f8r8cbpa1jv5zls945m8rm-docker-17.12.0-ce
checking for references to /build in /nix/store/j5nj3rs610f8r8cbpa1jv5zls945m8rm-docker-17.12.0-ce...
shrinking RPATHs of ELF executables and libraries in /nix/store/mgl18fw8y425lvb9aj3axc5fv2cqm988-docker-17.12.0-ce-man
gzipping man pages under /nix/store/mgl18fw8y425lvb9aj3axc5fv2cqm988-docker-17.12.0-ce-man/share/man/
patching script interpreter paths in /nix/store/mgl18fw8y425lvb9aj3axc5fv2cqm988-docker-17.12.0-ce-man
checking for references to /build in /nix/store/mgl18fw8y425lvb9aj3axc5fv2cqm988-docker-17.12.0-ce-man...
/nix/store/j5nj3rs610f8r8cbpa1jv5zls945m8rm-docker-17.12.0-ce
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.35539&attempt_id=fa277dd8-3f0d-423d-91b4-09f03c968bb7)

<details><summary>Partial log (click to expand)</summary><p>

```
cannot build derivation ‘/nix/store/lvzndaim8rpi7yj5mdxy8h41fnxp6vqa-etc.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/mba3mnc9ma5xaxgqsglvrkynralkzbvh-stage-1-init.sh.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/30nmr2r0dwkrh2h3bavs3w5wy3frf3rj-initrd.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/rfqffk0yjp6r50vn54qrs11wa142ml54-nixos-system-docker-18.03.git.2e82275.drv’: 5 dependencies couldn't be built
cannot build derivation ‘/nix/store/wdqy5iwl4ynv7b1p7b384kan49cplpn0-closure-info.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/38p1c16lks2n5b82fn4wfr90q3hgd44k-run-nixos-vm.drv’: 2 dependencies couldn't be built
cannot build derivation ‘/nix/store/h62wv68lmfl0hcqhac3c7pfv676960ab-nixos-vm.drv’: 2 dependencies couldn't be built
cannot build derivation ‘/nix/store/ki544y3jngfacicijmnlpz1ls01ylm3a-nixos-test-driver-docker.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/wxwgrf248qknhw30g05yjwlszjbg6zbk-vm-test-run-docker.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/wxwgrf248qknhw30g05yjwlszjbg6zbk-vm-test-run-docker.drv’ failed
```
</p></details>

srhb commented:
Error is unrelated. x86_64 test runs fine locally.

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.35539&attempt_id=3ca73008-80d3-42f4-b248-8b81f1feb3d5)

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/83jvxj8r3qkmi3fkfnzvfvyp66n6k10l-docker-17.12.0-ce
```
</p></details>

Mic92 commented:
That fixed all the docker tests, thanks.
